keep/rulesengine/rulesengine.py

```diff
@@ -156,6 +157,23 @@ def _run_cel_rules(
 
                                 send_created_event = incident.is_visible
 
+                            incident_id = incident
+                            # Incident might change till this moment
+                            for attempt in range(3):
+                                try:
+                                    session.add(incident)
```

Comment:
Can you explain the comment better? "incident might change till this moment" what does that mean, we add the incident here..?